.ci/publishNpm.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+set -e
+
+if [ -z "$VERSION" ]; then
+    echo "The environment variable 'VERSION' must be set."
+    exit 1
+fi
+
+# set the version of all packages in the workspace to $VERSION
+npm version -ws $VERSION
+
+# replace all dependencies from the current workspace that are marked with "*" with the version that gets published
+find . -regex "./packages/[A-Za-z-]*/package\.json$" -exec sed -i -e "s/\"\*\"/\"$VERSION\"/g" {} \;
```

Comment:
`-i` differs between GNU/Linux and MacOS implementations of `sed` (see [here](https://www.grymoire.com/Unix/Sed.html#uh-62h)). I'm not sure if that's still an issue, but I have this in the back of my head, since I've had a couple of scripts break when I ran them on a mac a couple of years ago.

If this script will only ever run in (linux) pipelines forget about it.

Reply:
I only tried it on mac so far, if it runs on linux like this it should work.